plugins/uefi-dbx/fu-uefi-dbx-common.c

```diff
@@ -31,6 +34,135 @@ fu_uefi_dbx_get_authenticode_hash(const gchar *fn, GError **error)
+	if (names == NULL)
+		return NULL;
+	for (guint i = 0; i < names->len; i++) {
+		const gchar *name = g_ptr_array_index(names, i);
+		efi_load_option *loadopt;
+		gsize var_data_size = 0;
+		efidp dp = NULL;
+		guint16 fp_list_len;
+		gssize dp_len;
+		g_autofree const gchar *text_path = NULL;
+		gsize text_path_len = 0;
+		g_autofree guint8 *var_data_tmp = NULL;
+		g_autoptr(GError) error_local = NULL;
+		g_auto(GStrv) tp_toks = NULL;
+		guint tp_toks_len = 0;
+
+		/* only BootXXXX */
+		if (!fu_uefi_dbx_entry_name_is_boot(name))
+			continue;
+
+		/* parse efivars data file */
+		if (!fu_efivar_get_data(guid,
+					name,
+					&var_data_tmp,
+					&var_data_size,
+					NULL,
+					&error_local)) {
+			g_debug("(%s) failed to get data: %s", name, error_local->message);
+			continue;
+		}
+
+		/* retrieve load option from data */
+		loadopt = (efi_load_option *)var_data_tmp;
+		if (!efi_loadopt_is_valid(loadopt, var_data_size)) {
+			g_debug("(%s) load option was invalid", name);
+			continue;
+		}
+
+		/* construct device path object */
+		dp = efi_loadopt_path(loadopt, var_data_size);
+
+		/* only Media Device Path type which holds the bootable image */
+		if (dp->type != 0x4)
+			continue;
+
+		/* know the length of device path list */
+		fp_list_len = efi_loadopt_pathlen(loadopt, var_data_size);
+
+		/* know the length of device path */
+		dp_len = efidp_format_device_path(NULL, 0, dp, fp_list_len);
+		if (dp_len < 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "(%s) could not parse device path",
+				    name);
+			return NULL;
+		}
+
+		/* allocate memory for device path in text */
+		text_path_len = dp_len + 1;
+		text_path = (gchar *)g_malloc(text_path_len);
+		if (!text_path) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "(%s) couldn't allocate memory for text path",
+				    name);
+			return NULL;
+		}
+
+		/* convert device path to text */
+		dp_len =
+		    efidp_format_device_path((char *)text_path, text_path_len, dp, fp_list_len);
+		if (dp_len < 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "(%s) incorrect device path",
+				    name);
+			return NULL;
+		}
+
+		/* Tokenize the dp text to extract the relative path of the
+		 * efi image, typically after the last appearance of '/'.
+		 *
+		 * Two examples for the text path after '/'
+		 * 1) File(\\EFI\\ubuntu\\shimx64.efi)
+		 * 2) \\EFI\\ubuntu\\shimx64.efi
+		 *
+		 * The second example is introduced in:
+		 * https://github.com/rhboot/efivar/commit/9a5e710
+		 */
+		tp_toks = g_strsplit(text_path, "/", -1);
+		tp_toks_len = g_strv_length(tp_toks);
+		if (tp_toks_len > 0) {
+			g_autoptr(GString) efip = g_string_new(tp_toks[tp_toks_len - 1]);
+			fu_string_replace(efip, "File(", "");
+			fu_string_replace(efip, ")", "");
+			fu_string_replace(efip, "\\", "/");
+
+			g_ptr_array_add(efi_paths, g_strdup(efip->str));
```

Comment:
As suggested, you should probably strip the last part of this so you get a directory instead.  Then you can list all the files in that directory and return a list of all the files.  That will make sure even if bootloaders are chained that it works properly.